src/components/Table/TableHead.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { visuallyHidden } from '@mui/utils';
+import type { AutoUIEntityPropertyDefinition } from '../../AutoUI';
+import type { CheckedState, Order } from './utils';
+import { Material } from '@balena/ui-shared-components';
+
+const {
+	Box,
+	TableCell,
+	TableHead: MaterialTableHead,
+	TableRow,
+	TableSortLabel,
+	Checkbox,
+} = Material;
+
+interface TableHeadProps<T> {
+	columns: Array<AutoUIEntityPropertyDefinition<T>>;
+	numSelected?: number;
+	checkedState?: CheckedState;
+	onRequestSort: (
+		event: React.MouseEvent<HTMLSpanElement, MouseEvent>,
+		column: AutoUIEntityPropertyDefinition<T>,
+	) => void;
+	onSelectAllClick?: (
+		selected: T[] | undefined,
+		allChecked: CheckedState,
+	) => void;
+	order: Order;
+	orderBy: string;
+	rowCount: number;
+}
+
+export const TableHead = <T extends object>({
+	columns,
+	onSelectAllClick,
+	order,
+	orderBy,
+	numSelected,
+	checkedState,
+	rowCount,
+	onRequestSort,
+}: TableHeadProps<T>) => {
+	return (
+		<MaterialTableHead>
+			<TableRow>
+				<TableCell data-table="table_cel__sticky_header" padding="checkbox">
+					<Checkbox
+						color="primary"
+						// some does not seems to work as expected
+						indeterminate={checkedState === 'some'}
+						checked={
+							(rowCount > 0 && numSelected === rowCount) ||
+							checkedState === 'all'
+						}
+						onChange={(_event, checked) => {
+							onSelectAllClick?.(undefined, checked ? 'all' : 'none');
+						}}
+						inputProps={{
+							'aria-label': 'select all',
+						}}
+					/>
+				</TableCell>
+				{columns.map((column) =>
+					column.selected ? (
+						<TableCell
+							key={column.key}
+							align="left"
+							padding="none"
+							sortDirection={orderBy === column.key ? order : false}
+							sx={{ whiteSpace: 'nowrap' }}
+						>
+							{column.sortable ? (
+								<TableSortLabel
+									active={orderBy === column.key}
+									direction={orderBy === column.key ? order : 'asc'}
```

Comment:
This seems a bit odd, how do we assume it's `asc`? What about `desc`?

Reply:
it's the default arrow position when you order by other fields. That's a default behaviour in tables
e.g if you order by Name then name follow the actual order, all the other columns will appear with the default arrow position which is "asc"